plutus-core/cost-model/print-cost-model/Main.hs

```diff
@@ -0,0 +1,237 @@
+
+-- | Print a linear function in readable form.  The string argument is
+-- supposed to represent the input to the function: x, y, y+z, etc.
+renderLinearFunction :: LinearFunction -> String -> String
+renderLinearFunction (LinearFunction intercept slope) var =
+    if intercept == 0 then stringOfMonomial slope var
+    else printf "%d + %s" intercept (stringOfMonomial slope var)
+        where stringOfMonomial s v =
+                  if s == 1 then unparen v  -- Just so we don't get things like 5 + (x+y).
+                  else if s == -1 then "-" ++ v
+                  else printf "%d*%s" s v
+                  -- Print the slope even if it's zero, so we know the
+                  -- function's not constant.
+              unparen v = if v /= "" && head v == '(' && last v == ')'
+                          then tail $ init v
+                          else v
+
+renderModel :: Model -> [String]
+renderModel =
+    \case
+     ConstantCost       n   -> [ printf "%d" n ]
+     AddedSizes         f   -> [ renderLinearFunction f "(x+y)" ]
+     MultipliedSizes    f   -> [ renderLinearFunction f "(x*y)" ]
+     MinSize            f   -> [ renderLinearFunction f "min(x,y)" ]
+     MaxSize            f   -> [ renderLinearFunction f "max(x,y)" ]
+     LinearCost         f   -> [ renderLinearFunction f "x" ]
+     LinearInX          f   -> [ renderLinearFunction f "x" ]
+     LinearInY          f   -> [ renderLinearFunction f "y" ]
+     LinearInZ          f   -> [ renderLinearFunction f "z" ]
+     SubtractedSizes    l c -> [ "if x>y"
+                               , printf "then %s" $ renderLinearFunction l "(x-y)"
+                               , printf "else %d" c
+                               ]
+     ConstAboveDiagonal c m -> [ "if x>y"
+                               , printf "then %s" $ intercalate "\n" (renderModel m)
+                               , printf "else %d" c
+                               ]
+     ConstBelowDiagonal c m -> [ "if x<y"
+                               , printf "then %s" $ intercalate "\n" (renderModel m)
+                               , printf "else %d" c
+                               ]
+     LinearOnDiagonal   f c -> [ "if x==y"
+                               , printf "then %s" $ renderLinearFunction f "x"
+                               , printf "else %d" c
+                               ]
+     -- ^ We're not properly indenting submodels in the above/below diagonal
+     -- cases, but at present our submodels all fit on one line (eg, constant or
+     -- linear).  It seems improbable that we'd ever have a submodel that
+     -- required more than one line because then we'd be dividing the plane up
+     -- into more than two regions.
+
+-- | Take a list of strings and print them line by line, the first with no extra
+-- spaces then the rest preceded by `width` spaces. The assumption is that we'll
+-- already have printed the first part of the first line.
+printListIndented :: Int -> [String] -> IO ()
```

Comment:
I wonder if it would make your life easier to use `prettyprinter`? Then you can use `align` and friends.

Reply:
I did consider that, but fundamentally I just couldn't be bothered.